topk-js/src/expr.rs

```diff
@@ -0,0 +1,343 @@
+      _ => unreachable!("lte can only be called on logical expressions"),
+    }
+  }
+
+  #[napi]
+  pub fn gt(&self, value: f64) -> Expr {
+    match &self.expr {
+      Expression::LogicalExpression { expr } => Self {
+        expr: Expression::LogicalExpression {
+          expr: expr.greater_than(&LogicalExpression::Literal {
+            value: Value::F64(value),
+          }),
+        },
+      },
+      _ => unreachable!("gt can only be called on logical expressions"),
+    }
+  }
+
+  #[napi]
+  pub fn gte(&self, value: f64) -> Expr {
+    match &self.expr {
+      Expression::LogicalExpression { expr } => Self {
+        expr: Expression::LogicalExpression {
+          expr: expr.greater_than_or_equal(&LogicalExpression::Literal {
+            value: Value::F64(value),
+          }),
+        },
+      },
+      _ => unreachable!("gte can only be called on logical expressions"),
+    }
+  }
+
+  #[napi]
+  pub fn add(&self, value: f64) -> Expr {
+    match &self.expr {
+      Expression::LogicalExpression { expr } => Self {
+        expr: Expression::LogicalExpression {
+          expr: expr.add(&LogicalExpression::Literal {
+            value: Value::F64(value),
+          }),
+        },
+      },
+      _ => unreachable!("add can only be called on logical expressions"),
+    }
+  }
+
+  #[napi]
+  pub fn subtract(&self, value: f64) -> Expr {
+    match &self.expr {
+      Expression::LogicalExpression { expr } => Self {
+        expr: Expression::LogicalExpression {
+          expr: expr.subtract(&LogicalExpression::Literal {
+            value: Value::F64(value),
+          }),
+        },
+      },
+      _ => unreachable!("subtract can only be called on logical expressions"),
+    }
+  }
+
+  #[napi]
+  pub fn multiply(&self, value: f64) -> Expr {
+    match &self.expr {
+      Expression::LogicalExpression { expr } => Self {
+        expr: Expression::LogicalExpression {
+          expr: expr.multiply(&LogicalExpression::Literal {
+            value: Value::F64(value),
+          }),
+        },
+      },
+      _ => unreachable!("multiply can only be called on logical expressions"),
+    }
+  }
+
+  #[napi]
+  pub fn divide(&self, value: f64) -> Expr {
+    match &self.expr {
+      Expression::LogicalExpression { expr } => Self {
+        expr: Expression::LogicalExpression {
+          expr: expr.divide(&LogicalExpression::Literal {
+            value: Value::F64(value),
+          }),
+        },
+      },
+      _ => unreachable!("divide can only be called on logical expressions"),
+    }
+  }
+
+  #[napi]
+  pub fn and(&self, other: Expr) -> Expr {
+    match &self.expr {
+      Expression::TextExpression { expr } => {
+        if let Expression::TextExpression { expr: other_expr } = &other.expr {
+          Self {
+            expr: Expression::TextExpression {
+              expr: expr.and(other_expr),
+            },
+          }
+        } else {
+          unreachable!("and can only be called on text expressions")
```

Comment:
this type of "error state" will disappear when `Expr` is split into multiple types